Added batchrename-0.2.0 recipe

axeld opened the pull request:
It's a Tracker add-on, and stand-alone application that I wrote a couple of years back, as I wasn't happy with the existing offerings. It's actual name is just 'rename', but that one is already taken by bbjimmy's add-on.

jmairboeck commented on haiku-apps/batchrename/batchrename-0.2.0.recipe:
```diff
@@ -0,0 +1,49 @@
+SOURCE_DIR="rename-b768c8525dbaf15b055ffd0b303b8c79a17d8c9a"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
```
Do Tracker add-ons even work for a secondary architecture?

axeld replied:
Oops, good point. I actually was too lazy to look into why it wouldn't build anymore with GCC2. Please hold off of the recipe until I fixed that, I completely forgot about this.

jmairboeck commented on haiku-apps/batchrename/batchrename-0.2.0.recipe:
```diff
@@ -0,0 +1,49 @@
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	icu74${secondaryArchSuffix}_devel
```
Usually, references to library names are used instead of package names, so something like `devel:libicuuc$secondaryArchSuffix >= 74`. But this is just a cosmetic change in this case.

axeld replied:
I actually only need the headers, but if your version work, then it would definitely be preferred, as I'm pretty agnostic about the actual version. Thanks for the suggestion!

jmairboeck replied:
If the version doesn't matter, you can also leave out the version constraint then. For gcc2, there is only icu-57 anyway (the package name has no version suffix). With the `devel:` syntax, you don't need to differentiate that then.

jmairboeck commented on haiku-apps/batchrename/batchrename-0.2.0.recipe:
```diff
@@ -0,0 +1,49 @@
+
+INSTALL()
+{
+	mkdir -p $binDir
```
`$binDir` has the secondary architecture appended, so it is `$prefix/bin/x86` on x86, and binaries are symlinked to `$prefix/bin` with `-x86` appended.

You should be using `$prefix/bin` here instead, or append `$secondaryArchSuffix` to the `cmd:rename` PROVIDES above (the former is preferred).

axeld replied:
Sorry, I don't get this. As it is now (as far as I understood), the binary "rename" would end up in bin/x86, and could therefore be started with just "rename" from a Terminal, right? That's at least what I wanted to achieve here.

augiedoggie replied:
If the command is in `$binDir` then the command would be `rename-x86` unless you used the `setarch x86` command and then you would be able to just use `rename`.  If you always want it to be available as `rename`, regardless of `setarch` status, then it needs to be in `$prefix/bin`

augiedoggie replied:
Of course, if you're going to fix gcc2 support then you won't be needing the secondary arch and using `$binDir` is fine in that case.

pulkomandy replied:
Axel is right here, the binary will end up in bin/x86 and be accessible in the path. There is no problem here.

However, for programs that don't build with gcc2, we have started to put the gcc13 version directly in bin/ anyways, as it is simpler and there is no reason to use the x86/ subdirectory in that case.

jmairboeck replied:
@pulkomandy Thanks for the clarification. I wasn't aware that `bin/x86` is in the PATH by default (I don't use 32 bit much). But the extra suffixed symlink would still be unneeded.

augiedoggie replied:
> @pulkomandy Thanks for the clarification. I wasn't aware that `bin/x86` is in the PATH by default (I don't use 32 bit much). But the extra suffixed symlink would still be unneeded.

It's not in my $PATH on either of my 32 bit VMs

axeld commented:
Are you, by any chance, have an idea why USER_SETTINGS_FILES would not work? When I left it in, the created .PackageInfo could not be parsed, and it would put quotes around the wrong part.

augiedoggie commented:
> Are you, by any chance, have an idea why USER_SETTINGS_FILES would not work? When I left it in, the created .PackageInfo could not be parsed, and it would put quotes around the wrong part.

The space in the path probably needs to be escaped.  `USER_SETTINGS_FILES="settings/pinc.rename\ settings"`

axeld commented:
I actually tried that, but with the input string of "settings/pinc.rename\ settings", it just put the '\' in the final string like this: {"settings/pinc-rename\" settings}

axeld commented:
Great, thanks, I will try that!

axeld commented:
I've updated the code base to compile with GCC 2 again, and fixed the remaining issues in the recipe, too. At least I hope so :-) It should be ready to go now.

augiedoggie commented:
I got a build failure when trying to build on x86_64 nightly.

```
RenameWindow.cpp: In constructor 'RenameWindow::RenameWindow(RenameSettings&)':
RenameWindow.cpp:141:48: error: call of overloaded 'SetVisibleItem(long int)' is ambiguous
  141 |         fCardView->CardLayout()->SetVisibleItem(0L);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
In file included from /boot/system/develop/headers/os/interface/LayoutBuilder.h:11,
                 from RenameWindow.cpp:23:
/boot/system/develop/headers/os/interface/CardLayout.h:19:57: note: candidate: 'void BCardLayout::SetVisibleItem(int32)'
   19 |                         void                            SetVisibleItem(int32 index);
      |                                                         ^~~~~~~~~~~~~~
/boot/system/develop/headers/os/interface/CardLayout.h:20:57: note: candidate: 'void BCardLayout::SetVisibleItem(BLayoutItem*)'
   20 |                         void                            SetVisibleItem(BLayoutItem* item);
      |                                                         ^~~~~~~~~~~~~~

```

Also, I don't think the old BeOS method of appending a shortcut like `-R` to the Tracker add-on filename works any longer.  Perhaps you could switch it to the "BEOS:default_shortcut" resource. When I built it on 32 bit, Tracker always showed  "Batch rename-R" with no shortcut.

augiedoggie commented:
Just another side note, the `util_linux` package also provides a `rename` command.

axeld commented:
I've renamed the executable to 'batchrename', too, and fixed the compiling issue on x86-64. However, I haven't got the time to really test those changes, and I haven't tried building it on x86-64 yet, either. I will probably find the time to do this tomorrow, though.

Begasus commented:
Could use a nice icon :)

Begasus commented on haiku-apps/batchrename/batchrename-0.2.0.recipe:
```diff
@@ -0,0 +1,48 @@
+COPYRIGHT="2019-2024 pinc Software"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/axeld/rename/archive/39a87c5d5127544f831e89fce91c2b33193cc103.tar.gz"
+CHECKSUM_SHA256="085bd2ceb244b6bd44e19903fe5efc08567bb05d42aa0e1dd88f554b756ec58a"
+SOURCE_DIR="rename-39a87c5d5127544f831e89fce91c2b33193cc103"
```
Suggestion, as this is a commit checkout we mostly use the variable srcGitRev to define the tag and rename the recipe to reflect that it's a git checkout: eg `batchrename-0.2.0~git.recipe`
```diff
-SOURCE_URI="https://github.com/axeld/rename/archive/39a87c5d5127544f831e89fce91c2b33193cc103.tar.gz"
-CHECKSUM_SHA256="085bd2ceb244b6bd44e19903fe5efc08567bb05d42aa0e1dd88f554b756ec58a"
-SOURCE_DIR="rename-39a87c5d5127544f831e89fce91c2b33193cc103"
+srcGitRev="39a87c5d5127544f831e89fce91c2b33193cc103"
+SOURCE_URI="https://github.com/axeld/rename/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="085bd2ceb244b6bd44e19903fe5efc08567bb05d42aa0e1dd88f554b756ec58a"
+SOURCE_FILENAME="batchrename-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="rename-$srcGitRev"
```

Begasus replied:
On a future update only the srcGitRev needs to be updated then.

axeld replied:
Thanks for testing, and the suggestions! That makes sense!

Begasus commented:
Build and Tracker addon working OK on 32bit and 64bit.
